envoy/common/key_value_store.h

```diff
@@ -21,8 +23,11 @@ class KeyValueStore {
    * Adds or updates a key:value pair in the store.
    * @param key supplies a key to add or update.
    * @param value supplies the value to set for that key.
+   * @param ttl optionally give a ttl>0. If it's an update, ttl will be updated as well. If no ttl,
```

Comment:
How about something like: `@param ttl optionally specifies a lifetime after which this entry will be removed. ttl must be greater than 0.` or some such?